demosplan/DemosPlanCoreBundle/Repository/CustomFieldConfigurationRepository.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Repository;
+
+use demosplan\DemosPlanCoreBundle\CustomField\CustomFieldInterface;
+use demosplan\DemosPlanCoreBundle\Entity\CustomFields\CustomFieldConfiguration;
+use demosplan\DemosPlanCoreBundle\Entity\Procedure\Procedure;
+use Doctrine\Common\Collections\ArrayCollection;
+use Exception;
+
+class CustomFieldConfigurationRepository extends CoreRepository
+{
+    /**
+     * Add Entity to database.
+     *
+     * @throws Exception
+     */
+    private function add(CustomFieldConfiguration $customFieldConfiguration): CustomFieldConfiguration
+    {
+        try {
+            $em = $this->getEntityManager();
+            $em->persist($customFieldConfiguration);
+            $em->flush();
+
+            return $customFieldConfiguration;
+        } catch (Exception $e) {
+            $this->logger->warning('CustomFieldConfiguration could not be added. ', [$e]);
+            throw $e;
+        }
+    }
+
+    public function findCustomFieldConfigurationByCriteria(string $sourceEntity, string $sourceEntityId, string $targetEntity)
+    {
```

Comment:
format and return type
```suggestion
    public function findCustomFieldConfigurationByCriteria(
        string $sourceEntity,
        string $sourceEntityId,
        string $targetEntity
    ): ?array {
```